tests/test_eval.py

```diff
@@ -0,0 +1,168 @@
+import math
+
+import pytest
+import torch
+from torch.utils.data import DataLoader
+
+from arcsf.data.data_module import EvalQADataset, get_data, qa_formatter_autoregression
+from arcsf.eval.metrics import (
+    conditional_probability,
+    eval_accuracy,
+    ks_test,
+    truth_ratio,
+)
+from arcsf.eval.utils import get_loss
+
+
+@pytest.fixture
+def dummy_data():
+    return get_data(
+        "tofu",
+        granularity="author",
+        stratified=True,
+        forget_random=False,
+        forgotten_author_fraction=1 / 3,
+        forgotten_fact_fraction=1 / 3,
+        random_seed=42,
+    )
+
+
+def test_accuracy():
+    """Tests that accuracy function is outputting correct values"""
+    # random predictions for test outputs
+    test_outputs_correct = torch.randn(100, 5)
+    # argmax predictions for test targets
+    test_targets = test_outputs_correct.argmax(dim=1)
+    # flip random predictions for incorrect predictions
+    test_outputs_incorrect = test_outputs_correct * -1
+
+    correct = eval_accuracy(test_outputs_correct, test_targets)
+    assert correct["eval_accuracy"] == pytest.approx(1.0)
+
+    incorrect = eval_accuracy(test_outputs_incorrect, test_targets)
+    assert incorrect["eval_accuracy"] == pytest.approx(0.0)
```

Comment:
These should be exact (try to only use `pytest.approx` where it's definitely needed):

```suggestion
    assert correct["eval_accuracy"] == 1.0

    incorrect = eval_accuracy(test_outputs_incorrect, test_targets)
    assert incorrect["eval_accuracy"] == 0.0
```

And a test of an intermediate value (between 0 and 1) would be a nice to have.

Reply:
added an intermediate test as well